pageserver/src/tenant/timeline/import_pgdata.rs

```diff
@@ -175,24 +97,151 @@ pub async fn doit(
                 .put_timeline_import_status(
                     timeline.tenant_shard_id,
                     timeline.timeline_id,
-                    // TODO(vlad): What about import errors?
                     ShardImportStatus::Done,
                 )
                 .await
                 .map_err(|_err| {
                     anyhow::anyhow!("Shut down while putting timeline import status")
                 })?;
         }
-        Some(ShardImportStatus::Error(err)) => {
+        ShardImportStatus::Error(err) => {
             info!(
                 "shard status indicates that the shard is done (error), skipping import {}",
                 err
             );
         }
-        Some(ShardImportStatus::Done) => {
+        ShardImportStatus::Done => {
             info!("shard status indicates that the shard is done (success), skipping import");
         }
     }
 
     Ok(())
 }
+
+async fn prepare_import(
+    timeline: &Arc<Timeline>,
+    storage: RemoteStorageWrapper,
+    cancel: &CancellationToken,
+) -> anyhow::Result<ControlFile> {
+    // Wipe the slate clean before starting the import as a precaution.
+    // This method is only called when there's no recorded checkpoint for the import
+    // in the storage controller.
+    info!("wipe the slate clean");
```

Comment:
```suggestion
    // Wipe the slate clean before starting the import as a precaution.
    // This method is only called when there's no recorded checkpoint for the import
    // in the storage controller.
    //
    // Note that this is plit-brain safe (two imports for same timeline shards running in
    // different generations) because we go through the usual deletion
    // path, including deletion queue.
    info!("wipe the slate clean");
```

Reply:
https://github.com/neondatabase/neon/pull/11862/commits/68143b07d6f4b4956e55caf7387b0e1937be259b